test/message/promise_handled_reject.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+const common = require('../common');
+
+// Flags: --expose-gc
+
+var p = new Promise(function(res, rej) {
+  consol.log('oops'); // eslint-disable-line no-undef
+});
+
+// Manually call GC due to possible memory contraints with attempting to
+// trigger it "naturally".
+setTimeout(common.mustCall(() => {
```

Comment:
It would probably be a good idea to add an `unhandledRejection` handler in this test too and assert that it gets called to help specify the difference in hooks through tests.